maintenance/cleanup.py

```diff
@@ -0,0 +1,176 @@
+            print value
+
+        for ann in image.listAnnotations():
+            if ann.OMERO_TYPE == omero.model.LongAnnotationI:
+                obj_ids_rating.append(ann.getId())
+                value = 'Will delete rating %s on image \
+                        %s of %s' % (ann.getId(), image.getId(), username)
+                print value
+            elif ann.OMERO_TYPE == omero.model.TagAnnotationI:
+                params = omero.sys.ParametersI()
+                params.add('imageId', wrap(image.getId()))
+                params.addId(ann.getId())
+                query = "select l.id from ImageAnnotationLink as \
+                        l where l.parent.id=:imageId AND l.child.id=:id"
+                linkIds = query_service.projection(query, params,
+                                                   conn.SERVICE_OPTS)
+                for linkId in linkIds:
+                    obj_ids_taglinks.append(linkId[0].getValue())
+                    value = 'Will delete link %s on image \
+                            %s of tag %s of %s' % (linkId[0].getValue(),
+                                                 image.getId(),
+                                                 ann.getId(), username)
+                    print value
+
+
+def delete_objs(conn):
+    print 'deleting', len(obj_ids_taglinks), ' tag links'
+    if len(obj_ids_taglinks) > 0:
+        conn.deleteObjects("ImageAnnotationLink", obj_ids_taglinks,
+                           deleteAnns=True, deleteChildren=False, wait=True)
+    print 'deleting %s rois' % len(obj_ids_rois)
+    if len(obj_ids_rois) > 0:
+        conn.deleteObjects("Roi", obj_ids_rois, deleteAnns=True,
+                           deleteChildren=True, wait=True)
+    print 'deleting %s ratings' % len(obj_ids_rating)
+    if len(obj_ids_rating) > 0:
+        conn.deleteObjects("LongAnnotation", obj_ids_rating, deleteAnns=True,
+                           deleteChildren=False, wait=True)
+
+
+def run(password, target, image, host, port):
+   
+    for i in range(1, 41):
+        username = "user-%s" % i
+        try:
+            conn = BlitzGateway(username, password, host=host, port=port)
+            conn.connect()
+            list_objs(conn, username, target)
+            cut_link(conn, username, image)
+        except Exception as exc:
+            print "Error while cleaning the objects: %s" % str(exc)
+        finally:
+            conn.close()
+
+    trainer_name = "trainer-1"
+    try:
+        conn = BlitzGateway(trainer_name, password, host=host, port=port)
+        conn.connect()
+        list_objs(conn, trainer_name, target)
+        cut_link(conn, trainer_name, image)
+        delete_objs(conn)
+    except Exception as exc:
+        print "Error while cleaning the objects: %s" % str(exc)
+    finally:
+        conn.close()
+
+
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('password')
+    parser.add_argument('dataset')
+    parser.add_argument('image')
```

Comment:
Please make the ``image`` argument optional.
Probably makes sense to make the ``dataset`` argument optional too